cli/src/main.rs

```
@@ -551,6 +557,35 @@ async fn main() -> eyre::Result<()> {
                 offer.reject().await?;
             }
         },
+        WormholeCommand::Completion { shell } => {
+            let mut cmd = WormholeCli::command();
+            let bin = cmd.get_name().to_string();
+
+            match shell {
+                shell @ Shell::Zsh => {
+                    // for zsh, we will wrap the output to make it easier to use
+                    // this way we can source it directly `source <(wormhole-rs completion zsh)`
+
+                    let mut out = Vec::new();
+                    clap_complete::generate(shell, &mut cmd, &bin, &mut out);
+
+                    let out = match String::from_utf8(out) {
+                        Ok(x) => format!(
+                            "compdef _{0} {0}\n _{0}() {{ {1} }}\n\nif [ \"$funcstack[1]\" = \"{0}\" ]; then\n   {0} \"$@\"\nfi",
+                            bin, x
+                        )
+                        .into_bytes(),
+                        Err(out) => out.into_bytes(),
```

That's a bit weird, why would we silently not process the output if it is not valid UTF-8? I think requiring UTF-8 is valid here, so you can just propagate the error up and then do business as usual.